CMakeLists.txt

```diff
@@ -848,18 +848,30 @@ endif()
 
 # Add all of the subdirectories, where we actually do work.
 
-# Include stdlib/ here because any targets defining IS_STDLIB
-# or TARGET_LIBRARY are added as implicit dependencies of the
-# swift-stdlib-* set of targets, but only after those targets have been
-# declared in stdlib's CMakeLists.
-# Currently, the only library that relies on this ordering is SwiftSyntax,
-# which lives in tools/.
+###############
+# PLEASE READ #
+###############
+#
+# We have to include stdlib/ before tools/.
+# Do not move add_subdirectory(stdlib) after add_subdirectory(tools)!                                             
+#
+# We must include stdlib/ before tools/ because stdlib/CMakeLists.txt
+# declares the swift-stdlib-* set of targets. These targets will then
+# implicitly depend on any targets declared with IS_STDLIB or
+# TARGET_LIBRARY.
+#
+# One such library that declares IS_STDLIB is SwiftSyntax, living in
+# tools/SwiftSyntax. It will silently fail to register its targets
```

Comment:
One more:

```
It will silently fail to register itstargets for building if the swift-stdlib-* targets, rending it unbuildable.
```

=>

```
If we include stdlib/ after tools/, the swift-stdlib-* set of targets will not have been generated yet causing the implicit dependency for SwiftSyntax to silently not be created. This then will cause SwiftSyntax to fail to build.
```

Your thoughts?

Reply:
👍